ts/modules/marker-clusters.ts

```diff
@@ -0,0 +1,1948 @@
+            x: number;
+            y: number;
+            point?: Point;
+            clusterZone?: ClusterZonesOptions;
+            clusterZoneClassName?: string;
+            pointsOutside?: Array<SplittedDataObject>;
+            pointsInside?: Array<SplittedDataObject>;
+        }
+        interface GroupMapOptionsObject extends SeriesOptionsType {
+            formatPrefix?: string;
+            userOptions?: string | number |
+            PointOptionsObject | (string | number)[] | null;
+            x?: number;
+            y?: number;
+        }
+        interface GroupMapObject {
+            options?: GroupMapOptionsObject;
+        }
+        interface ClusteredDataObject {
+            clusterElements: Array<ClusterAndNoiseObject>;
+            noiseElements: Array<ClusterAndNoiseObject>;
+            groupedXData: Array<number>;
+            groupedYData: Array<number>;
+            groupMap: Array<GroupMapObject>;
+            initMinX?: number;
+            initMaxX?: number;
+            initMinY?: number;
+            initMaxY?: number;
+        }
+        interface BaseClustersObject {
+            clusters: Array<ClusterAndNoiseObject>;
+            noise: Array<ClusterAndNoiseObject>;
+        }
+        interface Point {
+            isCluster?: boolean;
+            clusteredData?: Array<SplittedDataObject>;
+            clusterPointsAmount?: number;
+        }
+        interface Series {
+            clusters?: ClusteredDataObject;
+            clusterAlgorithms?: ClusterAlgorithmsDictionary;
+            clusteredSeriesData?: (Array<Point|null>|null);
+            gridValueSize?: number;
+            baseClusters?: (BaseClustersObject | null);
+            initMaxX?: number;
+            initMinX?: number;
+            initMaxY?: number;
+            initMinY?: number;
+            debugGridLines?: Array<SVGElement>;
+            dataMaxX?: number;
+            dataMinX?: number;
+            dataMaxY?: number;
+            dataMinY?: number;
+            getRealExtremes(): RealExtremesObject;
+            getGridOffset(): GridOffsetObject;
+            onDrillToCluster(
+                event: PointClickEventObject
+            ): void;
+            getClusterDistancesFromPoint(
+                clusters: Array<Highcharts.KmeansClusterObject>,
+                pointX: number,
+                pointY: number
+            ): Array<SortedKmeansClusterObject>;
+            getScaledGridSize(
+                options: Highcharts.LayoutAlgorithmOptions
+            ): number;
+            preventClusterCollisions(
+                props: PreventClusterCollisionsProps
+            ): PointPositionObject;
+            isValidGroupedDataObject(
+                groupedData: SplittedData
+            ): boolean;
+            getClusteredData(
+                groupedData: SplittedData,
+                options: MarkerClusterOptions
+            ): (ClusteredDataObject | boolean);
+            destroyClusteredData (): void;
```

Comment:
New module functions in generic classes should contain a mini doclet like
`/** @requires modules/marker-clusters */`.